examples/src/Form.tsx

```diff
@@ -4,49 +4,48 @@ import JSONTree from 'react-json-tree';
 import * as styles from './form.module.css';
 
 import { isRequired } from '../../src/validators';
+import { ValidationType } from '../../src/interfaces';
 
 export interface IProps extends IFormProps {}
 
-export const Form: React.SFC<IProps> = props => {
-    const { bindInput } = props.formMethods;
-    console.log('<<<', props);
-    return (
-        <div className={styles.view}>
-            <div className={styles.form}>
-                <h1>Login Form</h1>
-                <form name="login-form">
-                    <label className={styles.label}>
-                        Username:
-                        <input {...bindInput('username')} />
-                        <span className={styles.error}>
-                            {props.form.validationErrors['username']}
-                        </span>
-                    </label>
-                    <label className={styles.label}>
-                        Password:
-                        <input type="password" {...bindInput('password')} />
-                        <span className={styles.error}>
-                            {props.form.validationErrors['password']}
-                        </span>
-                    </label>
-                </form>
-            </div>
-            <div className={styles.json}>
-                <JSONTree data={props.form} />
-            </div>
+export const Form: React.SFC<IProps> = ({
+    form,
+    form: { validationErrors, validationWarnings },
+    formMethods: { bindInput }
+}) => (
+    <div className={styles.view}>
+        <div className={styles.form}>
+            <h1>Login Form</h1>
+            <form name="login-form">
+                <label className={styles.label}>
+                    Username:
+                    <input {...bindInput('username')} />
+                    <span className={styles.error}>{validationErrors['username']}</span>
+                    <span className={styles.warning}>{validationWarnings['username']}</span>
+                </label>
+                <label className={styles.label}>
+                    Password:
+                    <input type="password" {...bindInput('password')} />
+                    <span className={styles.error}>{validationErrors['password']}</span>
+                    <span className={styles.warning}>{validationWarnings['password']}</span>
+                </label>
+            </form>
         </div>
-    );
-};
+        <div className={styles.json}>
+            <JSONTree data={form} />
+        </div>
+    </div>
+);
 
 export const ConnectedForm = connectForm<IProps>(
```

Comment:
I wanted to ask you about this. My understanding is that `T` is the model, so in this example it would be something like...

```
interface IFormModel {
  username: string;
  password: string;
}

connectForm<IFormModel>(...
```

And in this example IProps is just IFormProps, which includes form, formMethods, and initialModel. That's why I removed it from the example. Am I missing something?

I had thought about raising a separate issue for this.

Reply:
no, I think you are right